src/usage/vector-types.md

```diff
@@ -24,16 +24,38 @@ We support three operators to calculate the distance between two `bvector` value
 - `<=>` (`bvector_cos_ops`): cosine distance, defined as $1 - \frac{\Sigma x_iy_i}{\sqrt{\Sigma x_i^2 \Sigma y_i^2}}$.
 - `<~>` (`bvector_jaccard_ops`): Jaccard distance, defined as $1 - \frac{|X\cap Y|}{|X\cup Y|}$.
 
-```sql
-
 Index can be created on `bvector` type as well.
 
 ```sql
-CREATE INDEX bvector ON items USING vectors (embedding bvector_l2_ops);
+CREATE INDEX your_index_name ON items USING vectors (embedding bvector_l2_ops);
 
 SELECT * FROM items ORDER BY embedding <-> '[1,0,1]' LIMIT 5;
 ```
 
+### Data type cast
+
+Cast between `vector`:
+```sql
+SELECT '[1, 0, 1]'::vector::bvector;
+SELECT '[1, 0, 1]'::bvector::vector;
+```
+
+From `ARRAY` or `real[]` to bvector:
+```sql
+SELECT ARRAY[1, 0, 1]::real[]::vector::bvector;
+```
+
+From string constructor:
+```sql
+SELECT '[1, 0, 1]'::bvector;
+```
+
+From binarize constructor:
+```sql
+SELECT binarize(ARRAY[-2, -1, 0, 1, 2]::real[]::vector);;
+-- [0, 0, 0, 1, 1]
+```
```

Comment:
Not all pairs of vector types are cast-able.

We guarantee:
* casts between `vector` and `real[]` exist
* casts between `vector` and `[prefix]vector` exist
* casts between `[prefix]vector` and `[prefix]vec[suffix]` exist

And we provide more functions to build a vector:
* `to_[prefix]vec[suffix]`
* `binarize`

Every user-defined postgresql type can be derived from `text` so we do not have to emphasize it.